src/lib/curl/base.h

```diff
@@ -55,6 +65,33 @@ typedef struct {
 	void			*uctx;			//!< Private data for the module using the API.
 } fr_curl_io_request_t;
 
+typedef struct {
+	char const		*tls_certificate_file;
+	char const		*tls_private_key_file;
+	char const		*tls_private_key_password;
+	char const		*tls_ca_file;
+	char const		*tls_ca_issuer_file;
+	char const		*tls_ca_path;
+	char const		*tls_random_file;
+	bool			tls_check_cert;
+	bool			tls_check_cert_cn;
+	bool			tls_extract_cert_attrs;
+} fr_curl_tls_t;
+
+static CONF_PARSER fr_curl_tls_config[] = {
```

Comment:
This should go in base.c, and the declaration should go in the header.  When you move it to base.c remove the `static` keyword.  When you add the declaration to base.h add the `extern` keyword.

Arrays like this should not go in headers, because the memory will be duplicated wherever they're used.  Moving the array into a source file and exporting the symbol, means there's only one copy, which is referenced by the code that uses it.

Reply:
If you're interested, data like this goes in the text memory segment of the library or executable. https://en.wikipedia.org/wiki/Code_segment.

Reply:
edit: If you're interested, memory like this goes in the data segment of the library or binary - https://en.wikipedia.org/wiki/Data_segment

Reply:
Grr, I was right first time, starts off in text, and gets copied to data.